### PR TITLE
Indexer v1 package SPECs

### PR DESCRIPTION
**Pull request by okynos**
|Related issue|
|---|
| #1099 |

## Description

Hello team,

We want to merge this branch into big-product branch
We have included:
- Indexer Debian build scripts
- Indexer RPM build scripts
- Debian 4.3.0 indexer SPECs
- RPM 4.3.0 indexer SPECs
- Improved file permissions on both systems.
- Improved installation procedure.
- Added default roles and users to indexer package from unattended development.
- Added wazuh-cert-tool and wazuh-passwords-tool
- A lot more...

## Tests

- Build the package in any supported platform
  - [x] Linux
  - [x] Package installation
  - [x] Package remove

- Tests for Linux RPM
  - [x] Build the package for x86_64
  - [x] `%files` section is correctly updated if necessary
- Tests for Linux deb
  - [x] Build the package for x86_64
  - [x] Check file permissions after installing the package

### CODE REVIEW

**Review by verdx — APPROVED** [review]
LGTM

**Review by alberpilot — CHANGES_REQUESTED** [review]
- Change the certificate names and configuration files to `demo` name, as Opendistro. 
- The config.yml by default must include (commented) 3 elasticsearch nodes in total. The node names must be `node-1`, `node-2` and `node-3`. Like this:
```
nodes:
  # Elasticsearch server nodes
  elasticsearch:
    name: node-1
    ip: <elasticsearch-node-ip>
    # name: <elasticsearch-node-name>
    # ip: <elasticsearch-node-ip>
    # name: <elasticsearch-node-name>
    # ip: <elasticsearch-node-ip>

  # Wazuh server nodes
  # Use node_type only with more than one Wazuh manager
  wazuh_servers:
    name: wazuh-server
    ip: <wazuh-manager-ip>
    # node_type: master
    # name: <wazuh-node-name>
    # ip: <wazuh-manager-ip>
    # node_type: worker

  # Kibana node
  kibana:
    name: wazuh-dashboard
    ip: <kibana-node-ip>
```

File: indexer/rpm/wazuh-indexer.spec [hunk 1]
```diff
@@ -0,0 +1,1138 @@
+# Wazuh package SPEC
+# Copyright (C) 2015-2022, Wazuh Inc.
```
[the remark this answers is not in the document]
Comment (okynos) [reply indexer/rpm/wazuh-indexer.spec]: v2

File: indexer/rpm/wazuh-indexer.spec [hunk 2]
```diff
@@ -0,0 +1,1138 @@
+# Wazuh package SPEC
```
[the remark this answers is not in the document]
Comment (okynos) [reply indexer/rpm/wazuh-indexer.spec]: 2ad2a4d3

File: stack/indexer/deb/debian/postinst [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+if command -v systemctl > /dev/null 2>&1; then
+    systemctl restart systemd-sysctl.service > /dev/null 2>&1 || true
+fi
+
+# Below codeblock is using the fact that postinst script is called with the most-recently configured version.
+# In other words, a fresh installed will be called like "postinst configure" with no previous version ($2 is null)
+if [ -z "$2" ]; then
+  # If $2 is null, this is an install	
+
+  # Setting Owner and group of own folders
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/deb/debian/postinst]: 5950942f

File: stack/indexer/deb/debian/postinst [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+if [ -f "/etc/default/wazuh-indexer" ]; then
+    . "/etc/default/wazuh-indexer"
+fi
+
+# to pick up /usr/lib/sysctl.d/wazuh-indexer.conf
+if command -v systemctl > /dev/null 2>&1; then
+    systemctl restart systemd-sysctl.service > /dev/null 2>&1 || true
+fi
+
+# Below codeblock is using the fact that postinst script is called with the most-recently configured version.
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/deb/debian/postinst]: ea35c66c

File: indexer/rpm/wazuh-indexer.spec [hunk 5]
```diff
@@ -0,0 +1,1138 @@
...
+
+if [ $1 = 1 ];then # Install
+    echo "%{USER} hard nproc 4096" >> /etc/security/limits.conf
+    echo "%{USER} soft nproc 4096" >> /etc/security/limits.conf
+    echo "%{USER} hard nofile 65535" >> /etc/security/limits.conf
+    echo "%{USER} soft nofile 65535" >> /etc/security/limits.conf
+    if [[ "$(uname -a)" =~ "centos6" ]] || [[ "$(uname -a)" =~ "el6" ]];then
+        echo "### Disabled filter install on CentOS 6 systems ###" >> %{CONFIG_DIR}/opensearch.yml
+        echo "bootstrap.system_call_filter: false" >> %{CONFIG_DIR}/opensearch.yml
+    fi
```
[the remark this answers is not in the document]
Comment (okynos) [reply indexer/rpm/wazuh-indexer.spec]: e7e9888c

File: indexer/rpm/wazuh-indexer.spec [hunk 6]
```diff
@@ -0,0 +1,1138 @@
...
+            /etc/init.d/wazuh-indexer restart > /dev/null 2>&1
+        fi
+
+    # Older Suse linux distributions do not ship with systemd
+    # but do not have an /etc/init.d/ directory
+    # this tries to stop the wazuh-indexer service on these
+    # as well without failing this script
+    elif [ -x /etc/rc.d/init.d/wazuh-indexer ] ; then
+        /etc/rc.d/init.d/wazuh-indexer restart > /dev/null 2>&1
+    fi
```
[the remark this answers is not in the document]
Comment (okynos) [reply indexer/rpm/wazuh-indexer.spec]: Nothing to do https://github.com/opensearch-project/OpenSearch/blob/8bf6fa6b4f525a764e90955c69c0efeecdc74a5b/distribution/packages/src/common/scripts/postinst#L93-L100

File: indexer/rpm/wazuh-indexer.spec [hunk 7]
```diff
@@ -0,0 +1,1138 @@
...
+
+    # Older Suse linux distributions do not ship with systemd
+    # but do not have an /etc/init.d/ directory
+    # this tries to stop the wazuh-indexer service on these
+    # as well without failing this script
+    elif [ -x /etc/rc.d/init.d/wazuh-indexer ] ; then
+        /etc/rc.d/init.d/wazuh-indexer stop > /dev/null 2>&1
+    else # Anything else
+        kill -15 `pgrep -f opensearch` > /dev/null 2>&1
+    fi
```
[the remark this answers is not in the document]
Comment (okynos) [reply indexer/rpm/wazuh-indexer.spec]: Nothing to do.

File: stack/indexer/deb/build_package.sh [hunk 8]
```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+# Wazuh package generator
+# Copyright (C) 2015-2021, Wazuh Inc.
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/deb/build_package.sh:4]: v2

File: stack/indexer/deb/builder.sh [hunk 9]
```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# Wazuh package builder
+# Copyright (C) 2015-2022, Wazuh Inc.
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/deb/builder.sh:4]: v2

File: stack/indexer/deb/debian/changelog [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+
+  * More info: https://documentation.wazuh.com/current/release-notes/
+
+ -- Wazuh, Inc <info@wazuh.com>  Wed, 5 Jan 2022 10:00:00 +0000
+
+wazuh-indexer (4.2.5-1) UNRELEASED; urgency=low
+
+  * More info: https://documentation.wazuh.com/current/release-notes/
+
+ -- Wazuh, Inc <info@wazuh.com>  Mon, 15 Nov 2021 16:47:07 +0000
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/deb/debian/changelog:11]: It has UNRELEASED tag, required by Debian guidelines.

File: stack/indexer/deb/debian/copyright [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+    GNU General Public License version 2.
+
+License:
+
+    GNU General Public License version 2.
+
+The Debian packaging is:
+
+    Copyright (C) 2015-2022 Wazuh, Inc <info@wazuh.com>
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/deb/debian/copyright:19]: v2

File: stack/indexer/deb/debian/postinst [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+        else
+            /etc/init.d/wazuh-indexer restart > /dev/null 2>&1
+        fi
+
+    # Older Suse linux distributions do not ship with systemd
+    # but do not have an /etc/init.d/ directory
+    # this tries to stop the wazuh-indexer service on these
+    # as well without failing this script
+    elif [ -x /etc/rc.d/init.d/wazuh-indexer ]; then
+        /etc/rc.d/init.d/wazuh-indexer restart > /dev/null 2>&1
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/deb/debian/postinst:102]: Nothing to do.

File: stack/indexer/deb/debian/postrm [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+export LOG_DIR="/var/log/${NAME}"
+
+#
+# This script is executed in the post-removal phase
+#
+#   On Debian,
+#       $1=remove    : indicates a removal
+#       $1=purge     : indicates an upgrade
+
+# source the default env file
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/deb/debian/postrm]: cfb4b3e7

File: stack/indexer/deb/debian/postinst [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+  # ---------------------------------------------------------------------------
+
+  # Setting Indexer default configuration
+  echo "${USER} hard nproc 4096" >> /etc/security/limits.conf
+  echo "${USER} soft nproc 4096" >> /etc/security/limits.conf
+  echo "${USER} hard nofile 65535" >> /etc/security/limits.conf
+  echo "${USER} soft nofile 65535" >> /etc/security/limits.conf
+
+else
+  # otherwise it is an upgrade
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/deb/debian/postinst]: cfb4b3e7

File: stack/indexer/deb/debian/rules [hunk 15]
```diff
@@ -0,0 +1,1035 @@
...
+# -*- makefile -*-
+# Sample debian/rules that uses debhelper.
+#
+# This file was originally written by Joey Hess and Craig Small.
+# As a special exception, when this file is copied by dh-make into a
+# dh-make output file, you may use that output file without restriction.
+# This special exception was added by Craig Small in version 0.37 of dh-make.
+#
+# Modified to make a template file for a multi-binary package with separated
+# build-arch and build-indep targets by Bill Allombert 2001
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/deb/debian/rules:11]: Maybe we could ask to Joey Hess and Craig Small.

File: stack/indexer/deb/debian/rules [hunk 16]
```diff
@@ -0,0 +1,1035 @@
...
+# Modified by Wazuh 
+# Copyright (C) 2015-2022, Wazuh Inc.
+#
+# This program is a free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public
+# License (version 2) as published by the FSF - Free Software
+# Foundation.
+
+# Uncomment this to turn on verbose mode.
+export DH_VERBOSE=1
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/deb/debian/rules:22]: Don't think so https://github.com/wazuh/wazuh-packages/blob/master/debs/SPECS/4.4.0/wazuh-manager/debian/rules#L14

File: stack/indexer/deb/debian/rules [hunk 17]
```diff
@@ -0,0 +1,1035 @@
...
+
+# -----------------------------------------------------------------------------
+
+override_dh_install:
+		rm -rf $(INSTALLATION_DIR)/
+
+		curl -o $(INDEXER_FILE) https://s3.amazonaws.com/warehouse.wazuh.com/stack/indexer/$(INDEXER_FILE)
+		tar -zvxf $(INDEXER_FILE)
+
+		# copy to target
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/deb/debian/rules]: 0d74a751

File: stack/indexer/rpm/docker/x86_64/Dockerfile [hunk 18]
```diff
@@ -0,0 +1,14 @@
+FROM centos:8
+
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/rpm/docker/x86_64/Dockerfile:2]: Downgraded to CentOS 7 in v2

File: indexer/rpm/wazuh-indexer.spec (hunk 2 above)
[the remark this answers is not in the document]
Comment (okynos) [reply indexer/rpm/wazuh-indexer.spec]: https://github.com/wazuh/wazuh-packages/commit/2ad2a4d3f73601e0c7374c2eb0155229f3e69147

File: stack/indexer/rpm/wazuh-indexer.spec [hunk 19]
```diff
@@ -0,0 +1,1139 @@
+# Wazuh package SPEC
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/rpm/wazuh-indexer.spec]: https://github.com/wazuh/wazuh-packages/commit/2ad2a4d3f73601e0c7374c2eb0155229f3e69147
1fcd99dd

File: stack/indexer/securityadmin_wrapper.sh [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+
+# Wazuh-indexer securityadmin wrapper
+# Copyright (C) 2022, Wazuh Inc.
+#
+# This program is a free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public
+# License (version 2) as published by the FSF - Free Software
+# Foundation.
+
+HOST="127.0.0.1"
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/securityadmin_wrapper.sh:11]: v2

File: stack/indexer/securityadmin_wrapper.sh [hunk 21]
```diff
@@ -0,0 +1,76 @@
...
+INSTALL_PATH="/usr/share/wazuh-indexer"
+CONFIG_PATH="/etc/wazuh-indexer"
+OPTIONS="-icl -nhnv"
+PORT="9800"
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/securityadmin_wrapper.sh:15]: v2

File: stack/indexer/securityadmin_wrapper.sh [hunk 22]
```diff
@@ -0,0 +1,76 @@
...
+# Copyright (C) 2022, Wazuh Inc.
+#
+# This program is a free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public
+# License (version 2) as published by the FSF - Free Software
+# Foundation.
+
+HOST="127.0.0.1"
+INSTALL_PATH="/usr/share/wazuh-indexer"
+CONFIG_PATH="/etc/wazuh-indexer"
```
[the remark this answers is not in the document]
Comment (okynos) [reply stack/indexer/securityadmin_wrapper.sh:13]: v2

**Pull request merged**